### DISCUSSION TITLE
Add timestamp for when a state was updated without changes

### DISCUSSION THREAD

**Discussion by Platiinius** [home-assistant/architecture#997] [General]
## Proposed Change
When a state receives an update with unchanged values, the update is dropped.
Instead of dropping the update entirely, store an additional timestamp (e.g. `last_refreshed`).
In order not to cause event spam the change event could also be dropped, but that is secondary.

## Rationale
Capturing the time this last happened would provide information about sensor availability and downtime detection.
It would also allow separate decisions for each use case without destroying existing automations, like changing sensor availability would do.
There is a basic implementation draft as a pull request in [core](https://github.com/home-assistant/core/pull/103001), it does not seem to be a very invasive change.
My personal use case is availability checking of various battery powered devices as well as connection status of the companion app.
Some more arguments are in the original post and the pull request.

## Related
- The original [forum post](https://community.home-assistant.io/t/add-a-last-refreshed-to-complement-last-changed-and-last-updated/620462) (not by me)
- The draft [pull request in `core`](https://github.com/home-assistant/core/pull/103001)

**Comment by frenck** [comment]
This now exists, last_reported.

../Frenck
